### PR TITLE
#166272344 Implement Comment Functionality

### PR DESCRIPTION
**Pull request by GerrardE**
#### What does this PR do?
Implements user comment CRUD Functionality

#### Description of Task to be completed?
- Write tests
- Define api controller and methods
- Define api routes

#### How should this be manually tested?
1. clone this repo
2. run "npm install"
3. run "npm start"
4. run "npm run test"

#### Any background context you want to provide?
`ALGORITHM`

Step 1: Start
Step 2: Take a new branch out of the develop
Step 3: Push to the remote repository
Step 4: Checkout of the local branch and pull to develop then checkout to the same branch
Step 5: Create database seeders for users and articles,
Step 6: Write tests for the post, get, put and delete route,
Step 7: Define the API Create, Read, Update and Delete methods in a CommentController,
Step 8: Define the route in the API routes folder
Step 9: Run tests and ensure they pass
Step 10: Push to the remote repository
Step 11: Stop.

#### What are the relevant pivotal tracker stories?
[#166272344](https://www.pivotaltracker.com/story/show/166272344)

#### Screenshots (if appropriate)
N/A

#### Questions:
N/A

### CODE REVIEW

File: server/controllers/comment.js [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+      const { slug } = req.params;
+      const userId = req.decoded.id;
+
+      const article = await Article.findOne({ where: { slug } });
+
+      if (!article) return Response.error(res, 404, 'Article does not exist');
+
+      const comment = await article.createComment({ userId, slug, body });
+
+      const payload = await Comment.findAll({
```
Comment (fidelisojeah) [inline server/controllers/comment.js]: abstract this thing o baba.

You would find that you are doing the same thing in multiple places.

Don't Repeat Yourself (DRY)

File: server/routes/api/articles.js [hunk 2]
```diff
@@ -2,9 +2,15 @@ import express from 'express';
 import ArticleController from '@controllers/articles';
 import Token from '@helpers/Token';
 import trim from '@middlewares/trim';
+import CommentController from '@controllers/comment';
 
 const articlesRouter = express.Router();
 
+articlesRouter.post('/:slug/rate', trim, Token.authorize, ArticleController.rate);
+articlesRouter.post('/:slug/comment', Token.authorize, trim, CommentController.create);
```
Comment (fidelisojeah) [inline server/routes/api/articles.js]: please use plural forms `/comments`

**Review by fidelisojeah — DISMISSED** [review]

File: server/controllers/comment.js (hunk 1 above)
Comment (GerrardE) [reply server/controllers/comment.js]: on it now

File: server/routes/api/articles.js (hunk 2 above)
Comment (GerrardE) [reply server/routes/api/articles.js]: Feedback implemented

File: server/controllers/comment.js (hunk 1 above)
Comment (GerrardE) [reply server/controllers/comment.js]: Feedback implemented

**Review by despeauxz — APPROVED** [review]

**Review by despeauxz — DISMISSED** [review]
Nice job on the task, you covered a lot on Comment CRUD Operations

File: server/controllers/comments.js [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+   * @memberof CommentController
+   * @static
+   */
+  static async getComments(req, res, next) {
+    try {
+      const { article } = req;
+
+      const payload = await comments(article.id);
+
+      if (payload) {
```
Comment (fidelisojeah) [inline server/controllers/comments.js]: You probably should return the query ignoring the if statement

**Review by fidelisojeah — COMMENTED** [review]
Good job.

Please ensure that your endpoint always returns something (even if that something is an error

**Review by EMacco — APPROVED** [review]
Good job

File: server/controllers/comments.js (hunk 3 above)
Comment (GerrardE) [reply server/controllers/comments.js]: True that. I have implemented your feedback.

**Review by despeauxz — APPROVED** [review]
Nice job with the feedback implementation

**Review by fidelisojeah — APPROVED** [review]

**Pull request merged**